psycopg_pool/psycopg_pool/abc.py

```diff
@@ -34,3 +32,12 @@
     Callable[["AsyncConnectionPool[Any]"], None],
     Callable[["AsyncConnectionPool[Any]"], Awaitable[None]],
 ]
+
+# Types of the connection parameters
+ConninfoParam: TypeAlias = Union[str, Callable[[], str]]
+AsyncConninfoParam: TypeAlias = Union[str, Callable[[], Awaitable[str]]]
+
+KwargsParam: TypeAlias = Union[dict[str, Any], Callable[[], dict[str, Any]]]
+AsyncKwargsParam: TypeAlias = Union[
+    dict[str, Any], Callable[[], Awaitable[dict[str, Any]]]
+]
```

Comment:
These types are not covering the case that an async pool might still use a non-async function to get their credentials. In a previous suggestion I suggested to make the async version to be the union of the sync version and an async function. The types might also be defined this way:

```suggestion
ConninfoParam: TypeAlias = Union[str, Callable[[], str]]
AsyncConninfoParam: TypeAlias = Union[
    str,
    Callable[[], str],
    Callable[[], Awaitable[str]],
]

KwargsParam: TypeAlias = Union[dict[str, Any], Callable[[], dict[str, Any]]]
AsyncKwargsParam: TypeAlias = Union[
    dict[str, Any],
    Callable[[], dict[str, Any]],
    Callable[[], Awaitable[dict[str, Any]]],
]
```